utils/lit/lit/formats/taef.py

```diff
@@ -65,17 +64,20 @@ def getTaefTests(self, dll_path, litConfig, localConfig):
         if litConfig.debug:
             litConfig.note('searching taef test in %r' % dll_path)
         
-        select_arg = str.format('/select:"{}"', self.select_filter)
+        cmd = [self.te, dll_path, "/list", "/select:", self.select_filter]
 
-        cmd = [self.te, dll_path, "/list", select_arg]
         try:
-            lines,_,_ = executeCommandForTaef(cmd)
+            lines,err,exitCode = executeCommandForTaef(cmd)
             # this is for windows
             lines = lines.replace('\r', '')
             lines = lines.split('\n')
 
         except:
-            litConfig.error("unable to discover taef tests in %r, using %s" % (dll_path, self.te))
+            litConfig.error("unable to discover taef tests in %r, using %s. exeption encountered." % (dll_path, self.te))
+            raise StopIteration
+
+        if exitCode:
+            litConfig.error("unable to discover taef tests in %r, using %s. error: %s." % (dll_path, self.te, err))
```

Comment:
I chose to raise an exception independently to avoid the potentially confusing error that an exception was encountered while handling the exception.  It also makes printing the error message only when present easier